### PR TITLE
update to new producer

### PR DESCRIPTION
**Pull request by talevy**
New 0.8.2 Producer introduced new configuration options. This plugin reflects the 
switch to the new producer. This change is not backwards compatible with the previous version.

### CODE REVIEW

**Comment by suyograo** [comment]
@talevy we should block this one until we have the warning ready: https://github.com/elastic/logstash/issues/3383

**Comment by joekiller** [comment]
I guess this can move forward as of 6 days ago :)

**Comment by talevy** [comment]
@joekiller you're right! please comment on the new plugin if you see any things you wish to have updated.

**Comment by joekiller** [comment]
I think it is a pretty straightforward upgrade!

Perhaps I'll finish that benchmark project and we can get am idea of how it performs vs the Scala producer.

**Comment by suyograo** [comment]
LGTM

File: lib/logstash/outputs/kafka.rb [hunk 1]
```diff
@@ -20,132 +20,114 @@
...
+    @producer = Kafka::KafkaProducer.new(options)
     @producer.connect
 
-    @logger.info('Registering kafka producer', :topic_id => @topic_id, :broker_list => @broker_list)
+    @logger.info('Registering kafka producer', :topic_id => @topic_id, :bootstrap_servers => @bootstrap_servers)
 
     @codec.on_event do |event, data|
       begin
-        @producer.send_msg(event.sprintf(@topic_id),@partition_key,data)
+        @producer.send_msg(event.sprintf(@topic_id), nil, nil, data)
```
Comment (suyograo) [inline lib/logstash/outputs/kafka.rb]: just curious: we can't have a per message serializer now? i.e. picking it at the time of sending messages?

Comment (talevy) [inline lib/logstash/outputs/kafka.rb]: ah, indeed we can and should expose this.

```
config :partition, :validate => string
config :message_key => string
```

not sure how useful `partition` would be, but `message_key` could be used appropriately from within the config.

Comment (suyograo) [inline lib/logstash/outputs/kafka.rb]: yeah lets expose `message_key`!

**Comment by suyograo** [comment]
LGTM, @talevy and I spoke about releasing a beta version. This has a dependency on https://github.com/elastic/logstash/issues/3449 which adds support for installing prerelease packages on LS core. #3449 will be fixed in 1.5.3 so lets hold off a release until then

**Comment by joekiller** [comment]
Looks like we are good to go now. @talevy @suyograo

**Comment by suyograo** [comment]
@joekiller this code is good to go, but we would not be able to publish a new version of this plugin immediately. Our docs infra does not yet support versions for plugins, so we are working to fix that. Once this is ready (soon) we'll be able to merge this in, and publish a new version

**Pull request closed without merging**